src/dataflow/find_command_result.rs

```diff
@@ -0,0 +1,182 @@
+                        ),
+                    }
+                } else {
+                    match lockfile.get_module(
+                        &lockfile_command.package_name,
+                        &lockfile_command.package_version,
+                        &lockfile_command.module,
+                    ) {
+                        Ok(lockfile_module) => {
+                            let path = PathBuf::from(&lockfile_module.entry);
+                            FindCommandResult::CommandFound(
+                                path,
+                                lockfile_command.main_args.clone(),
+                                lockfile_module.name.clone(),
+                            )
+                        }
+                        Err(e) => FindCommandResult::Error(e),
+                    }
+                }
+            }
+            Err(e) => FindCommandResult::Error(e),
+        }
+    }
+
+    fn find_command_in_lockfile<S: AsRef<str>>(command_name: S, lockfile: Lockfile) -> Self {
+        match lockfile.get_command(command_name.as_ref()) {
+            Ok(lockfile_command) => {
+                match lockfile.get_module(
+                    &lockfile_command.package_name,
+                    &lockfile_command.package_version,
+                    &lockfile_command.module,
+                ) {
+                    Ok(lockfile_module) => {
+                        let path = PathBuf::from(&lockfile_module.entry);
+                        FindCommandResult::CommandFound(
+                            path,
+                            lockfile_command.main_args.clone(),
+                            lockfile_module.name.clone(),
+                        )
+                    }
+                    Err(_e) => {
+                        FindCommandResult::CommandNotFound(command_name.as_ref().to_string())
+                    }
+                }
+            }
+            Err(_e) => FindCommandResult::CommandNotFound(command_name.as_ref().to_string()),
+        }
+    }
+
+    pub fn find_command_in_directory<P: AsRef<Path>, S: AsRef<str>>(
+        directory: P,
+        command_name: S,
+    ) -> Self {
+        let manifest_result = ManifestResult::find_in_directory(&directory);
+        let lockfile_result = LockfileResult::find_in_directory(&directory);
+        match (manifest_result, lockfile_result) {
+            (ManifestResult::ManifestError(e), _) => return FindCommandResult::Error(e.into()),
+            (_, LockfileResult::LockfileError(e)) => return FindCommandResult::Error(e.into()),
+            (ManifestResult::NoManifest, LockfileResult::NoLockfile) => {} // continue
+            (ManifestResult::NoManifest, LockfileResult::Lockfile(l)) => {
+                return Self::find_command_in_lockfile(command_name, l)
+            }
+            (ManifestResult::Manifest(_m), LockfileResult::NoLockfile) => {
+                panic!("Manifest exists, but lockfile not found!")
```

Comment:
Is this previously verified in the code?  If so, then I think panic is fine, otherwise then I have a preference for not panicking, we can discuss it though!